crates/nu-cmd-extra/src/extra/bits/and.rs

```diff
@@ -42,51 +70,131 @@ impl Command for BitsAnd {
         input: PipelineData,
     ) -> Result<PipelineData, ShellError> {
         let head = call.head;
-        let target: i64 = call.req(engine_state, stack, 0)?;
+        let target: Value = call.req(engine_state, stack, 0)?;
+        let endian = call.get_flag::<Value>(engine_state, stack, "endian")?;
+
+        let little_endian = match endian {
+            Some(val) => {
+                let span = val.span();
+                match val {
+                    Value::String { val, .. } => match val.as_str() {
+                        "native" => cfg!(target_endian = "little"),
+                        "little" => true,
+                        "big" => false,
+                        _ => {
+                            return Err(ShellError::TypeMismatch {
+                                err_message: "Endian must be one of native, little, big"
+                                    .to_string(),
+                                span,
+                            })
+                        }
+                    },
+                    _ => false,
+                }
+            }
+            None => cfg!(target_endian = "little"),
+        };
 
-        // This doesn't match explicit nulls
         if matches!(input, PipelineData::Empty) {
             return Err(ShellError::PipelineEmpty { dst_span: head });
         }
-        input.map(
-            move |value| operate(value, target, head),
-            engine_state.ctrlc.clone(),
-        )
+
+        let args = Arguments {
+            target,
+            little_endian,
+        };
+
+        operate(action, args, input, head, engine_state.ctrlc.clone())
     }
 
     fn examples(&self) -> Vec<Example> {
         vec![
             Example {
-                description: "Apply bits and to two numbers",
+                description: "Apply logical and to two numbers",
```

Comment:
Correct me if I'm wrong, but I think these are all bitwise ands, not logical ands? I'm not sure why the existing example below said "Apply logical and to a list of numbers"

Reply:
thanks for catching 👍